packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -218,6 +238,58 @@ export class CohereChatLanguageModel implements LanguageModelV1 {
                 return;
               }
 
+              case 'tool-calls-chunk': {
+                if (value.tool_call_delta) {
+                  const { index } = value.tool_call_delta;
+
+                  if (!(index in toolCalls) || toolCalls[index] === undefined) {
```

Comment:
would `toolCalls[index] === undefined` not be sufficient?

Reply:
updated!